torch/csrc/jit/codegen/cuda/compute_at.cpp

```diff
@@ -715,6 +725,27 @@ void ComputeAt::updateSiblings() {
   }
 }
 
+void ComputeAt::updateInputProduceAts() {
+  std::unordered_set<TensorView*> consumers_to_check;
+
+  // Find all tensor views that may have been modified
+  auto chains = producer_use_chains_;
+  if (common_consumer_ != nullptr) {
+    chains = tvChains(
+        DependencyCheck::getAllDependencyChains(producer_, common_consumer_));
+  }
+
+  for (auto chain : chains) {
+    if (chain.size() > 1) {
+      consumers_to_check.emplace(chain[1]);
```

Comment:
Here, if I understand this correctly, we can also add `if (chain[0]->isFusionInput())`, correct?